### DISCUSSION TITLE
Will Jax be dropping Python 3.7 as per NEP 29?

### DISCUSSION THREAD

**Discussion by NeilGirdhar** [jax-ml/jax#9307] [General] [answered]
Jax appears to follow [NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html) along with many other Python projects in the scientific community.  Will Jax be dropping Python 3.7 soon?

**Comment by jakevdp** [comment, marked as answer]
In general we follow NEP 29, but dropping Python 3.7 is going to be delayed because it remains a requirement for some of the project's stakeholders.

**Reply by hawkinsp** [reply]
Note: this is as documented: https://jax.readthedocs.io/en/latest/deprecation.html

"JAX may support older versions of Python and NumPy, but support for older versions may be dropped at any time."

I'd expect we will drop Python 3.7 support some time in the next few months.

**Reply by NeilGirdhar** [reply]
Thanks, @hawkinsp, that's nice to know.
